apps.kruise.io/StatefulSet/health.lua

```diff
@@ -0,0 +1,23 @@
+hs={ status = "Progressing", message = "Waiting for initialization" }
+
+if obj.status ~= nil then
+
+    if obj.metadata.generation == obj.status.observedGeneration then
+
+        if obj.status.updatedReplicas == obj.status.replicas then
```

Comment:
advance statefulset status also contains UpdatedAvailableReplicas, so use the field to match against status.replicas